scripts/qesap/test/e2e/test.sh

```diff
@@ -149,18 +170,86 @@ set +e
+test_step "[test_3.yaml] Run dryrun Terraform"
+# correct execution of terraform in dryrun mode:
+# 1. test is checking for 0 exit code
+# 2. for skipping generation of the terraform.tfstate
+# Test is using an empty main.tf placed in the right provider folder
+mkdir -p "${TEST_PROVIDER}"
+touch "${TEST_PROVIDER}/main.tf"
+THIS_LOG="${QESAPROOT}/test_3_terraform.log"
+qesap.py --verbose -b ${QESAPROOT} -c test_3.yaml --dryrun terraform || test_die "test_3.yaml fail on dryrun terraform"
+TEST_TERRAFORM_TFSTATE="${TEST_PROVIDER}/terraform.tfstate"
+[[ ! -f "${TEST_TERRAFORM_TFSTATE}" ]] || test_die "File ${TEST_TERRAFORM_TFVARS} has not to be generated in dryrun mode!"
+
+test_split
+echo "Run the script again collecting the output"
+qesap.py -b ${QESAPROOT} -c test_3.yaml --dryrun terraform  |& tee "${THIS_LOG}"
+for t_step in init plan apply; do
+  echo "***Detect ${t_step} command***"
+  grep -qE "terraform.*${t_step}" \
+    "${THIS_LOG}" || test_die "test_3.yaml terraform dryrun does not have expected ${t_step} command in the output"
+done
+terraform_invocations=$(cat "${THIS_LOG}" | wc -l)
+[[ $terraform_invocations -eq 3 ]] || test_die "terraform dryrun does not emit exactly 3 commands but ${terraform_invocations}"
+[[ ! -f "${TEST_TERRAFORM_TFSTATE}" ]] || test_die "File ${TEST_TERRAFORM_TFVARS} has not to be generated"
+rm ${THIS_LOG}
+rm -rf "${TEST_PROVIDER}"
+
+#######################################################################
 test_step "[test_3.yaml] Run Terraform"
 # correct execution of terraform: test is checking for 0 exit code
 # and for the generation of the terraform.tfstate
 # terraform.tfstate is directly created by the terraform executable
 # Test is using an empty main.tf placed in the right provider folder
-touch "${TEST_PROVIDER}/main.tf"
+mkdir -p "${TEST_PROVIDER}"
+cp main_local.tf "${TEST_PROVIDER}/main.tf"
 qesap.py --verbose -b ${QESAPROOT} -c test_3.yaml terraform || test_die "test_3.yaml fail on terraform"
-TEST_TERRAFORM_TFSTATE="${TEST_PROVIDER}/terraform.tfstate"
 test_file "${TEST_TERRAFORM_TFSTATE}"
+test_file "${TEST_PROVIDER}/foo.bar"
+# do not delete tfstate to lave them fo rthe next test
+
+#######################################################################
+test_step "[test_3.yaml] Run dryrun Terraform destroy"
+# correct execution of terraform destroy in dryrun mode:
+# 1. test is checking for 0 exit code
+THIS_LOG="${QESAPROOT}/test_3_terraform.log"
+qesap.py --verbose -b ${QESAPROOT} -c test_3.yaml --dryrun terraform -d || test_die "test_3.yaml fail on dryrun terraform destroy"
+
+test_split
+echo "Run the script again collecting the output"
+qesap.py -b ${QESAPROOT} -c test_3.yaml --dryrun terraform -d |& tee "${THIS_LOG}"
+for t_step in destroy; do
+  echo "***Detect ${t_step} command***"
+  grep -qE "terraform.*${t_step}" \
+    "${THIS_LOG}" || test_die "test_3.yaml terraform dryrun does not have expected ${t_step} command in the output"
+done
```

Comment:
Why do we need a loop for a single test step? Is it just thinking ahead?

Reply:
cut and paste, let me fix it